### PR TITLE
Alter TEMP folder when running in CI

### PR DESCRIPTION
**Pull request by jeffkl**
Hoping this will insulate us a little bit from problems we're seeing in CI where things in TEMP are messed up.

### CODE REVIEW

File: cibuild.sh [hunk 1]
```diff
@@ -14,6 +14,8 @@ usage()
 }
 
 restoreBuildTools(){
+    # Work around issue where NuGet can't restore  because it can't get a lock
+
```
Comment (rainersigwald) [inline cibuild.sh]: How does this work around?

Comment (jeffkl) [inline cibuild.sh]: By placing TEMP under WORKSPACE, then we know that Jenkins will clean it up.  This at least gives us a clean TEMP folder for each build.

Comment (rainersigwald) [inline cibuild.sh]: But this is just a comment. What does it apply to?

Comment (jeffkl) [inline cibuild.sh]: Whooops I typed this comment, added code, and then moved the code without the comment :sad:

**Comment by jeffkl** [comment]
It turns out that the the workspace path has `---` in it which breaks some of our tests which try to write the path as an XML comment.  So I'm going to just do this for non-Windows for now.

What we should do in the long run is have a known TEMP path that gets deleted before and after the CI build.

**Pull request merged**